stdlib/public/TensorFlow/TensorGroup.swift

```diff
@@ -52,6 +53,22 @@ public extension TensorGroup {
   }
 }
 
+/// A protocol for types that can be used as tensor operation inputs and
+/// outputs. When a TensorGroup is used as an input, it gets passed to the
+/// tensor operation as an input list whose elements are the tensor fields of
+/// the type. When a TensorGroup is used as an output, it gets initialized
+/// with its tensor fields set to the tensor operation's tensor outputs.
+///
+/// TODO: Add a derived conformance to TensorGroup so that users don't have
+/// to write the conformance themselves.
+public protocol TensorGroup : InputTensorGroup & OutputTensorGroup {}
+
+public extension TensorGroup {
```

Comment:
I think that a default for this is more confusing than useful, because it sort of makes OutputTensorGroup partially conform to InputTensorGroup. I think things will be easier to think about if they are totally separate things without any overlap.